docs/kratos/manage-identities/60_external-id.mdx

```diff
@@ -0,0 +1,231 @@
+Content-Type: application/json
+
+{
+  "schema_id": "default",
+  "traits": {
+    "email": "user@example.com"
+  },
+  "external_id": "customer-12345"
+}
+```
+
+### Get identity by external id
+
+```http
+GET /admin/identities/by/external/customer-12345
+```
+
+#### Optional query parameter
+
+- `include_credential=password,oidc,...` — Include specific credentials in the response.
+
+#### Example:
+
+```http
+GET /admin/identities/by/external/customer-12345?include_credential=password
+```
+
+#### Response:
+
+```json
+{
+  "id": "uuid-abc123",
+  "external_id": "customer-12345",
+  "traits": {
+    "email": "user@example.com"
+  },
+  "credentials": {
+    "password": { ... }
+  }
+}
+```
+
+#### Error responses:
+
+- `404` – Identity not found.
+- `409` – Duplicate `external_id` on creation.
+- `400` – Invalid request structure.
+
+---
+
+### Other APIs
+
+There are no other additional APIs that support `external_id`, for the APIs that require a Kratos `identity_id` you will have to
+use the `Get identity by external id` API above and use the identity id from there.
+
+## JWT configuration
+
+### Jsonnet example
+
+When [tokenizing sessions](../../identities/session-to-jwt-cors), `external_id` is available in the session context:
+
+```jsonnet
+local claims = std.extVar('claims');
+local session = std.extVar('session');
+
+{
+  claims: {
+    iss: claims.iss + "/additional-component",
+    schema_id: session.identity.schema_id,
+    external_id: session.identity.external_id,
+    session: session,
+  }
+}
+```
+
+### Token behavior with `external_id`
+
+If `subject_source` is set to `external_id` in the tokenizer template, the JWT's `sub` claim becomes:
+
+```json
+{
+  "sub": "customer-12345"
+}
+```
+
+:::warning
+
+If `external_id` is missing, tokenization will fail.
+
+:::
+
+---
+
+## Migration guide
+
+If you're migrating from ForgeRock or a similar IAM that uses domain-specific IDs, you can import identities into Kratos with
+their `external_id` using the [batch API](https://www.ory.sh/docs/reference/api#tag/identity/operation/batchPatchIdentities).
+
+### Use: `PATCH /admin/identities`
+
+#### Example: basic import
+
+```json
+[
+  {
+    "schema_id": "default",
+    "external_id": "customer-001",
+    "traits": {
+      "email": "alice@example.com"
+    }
+  }
+]
+```
+
+#### Example: with pre-hashed password
+
+```json
+[
+  {
+    "schema_id": "default",
+    "external_id": "customer-002",
+    "traits": {
+      "email": "bob@example.com"
+    },
+    "credentials": {
+      "password": {
+        "config": {
+          "hashed_password": "$2b$12$abc123..." // bcrypt hash
+        }
+      }
+    }
+  }
+]
+```
+
+### Migration tips
+
+- Pre-hash passwords to avoid timeouts
+- Validate external_id uniqueness before import
+- Import in smaller batches (≤ 200 identities if using plaintext passwords)
+
+---
+
+## Troubleshooting
+
+| Error Code           | Description                                              |
```

Comment:
The torubleshooting section is for importing users or in general? What do you think of adding some description explaining for which area/part it reflects to?
`Tokenization failure` I'd assume also comes with HTTP Response like 500?

Reply:
Good question, I added it thinking of the top errors/issues most users might run into here (both migration + external_id specific ones).

Reply:
Tweaked it to make it more descriptive, please have a look